object.c

```diff
@@ -3236,15 +3236,22 @@ ALWAYS_INLINE(static VALUE rb_to_integer_with_id_exception(VALUE val, const char
 static inline VALUE
 rb_to_integer_with_id_exception(VALUE val, const char *method, ID mid, int raise)
 {
+    // when !raise, we need to pop the lazily pushed frame.
```

Comment:
It seems to be popped on L3254 when `raise` too. But is the case unreachable because `try_to_int` raises?

Reply:
I edited the comment to clarify that it's not really related to the parameter, but whether an exception is raised (we need to pop if the function returns normally)